### PR TITLE
Documenting fields for Pubsub schema transform in Syndeo

### PR DESCRIPTION
**Pull request by pabloem**

### CODE REVIEW

**Review by andreigurau — DISMISSED** [review]

File: syndeo-template/src/main/java/com/google/cloud/syndeo/transforms/pubsub/SyndeoPubsubReadSchemaTransformProvider.java [hunk 1]
```diff
@@ -178,12 +184,27 @@ public PCollectionRowTuple expand(PCollectionRowTuple input) {
...
     public abstract @Nullable String getSubscription();
 
-    public abstract String getDataFormat(); // AVRO, JSON
+    @SchemaFieldDescription(
+        "The encoding format for the data stored in Pubsub. Valid options are: "
+            + VALID_FORMATS_STR)
+    public abstract String getFormat(); // AVRO, JSON
 
+    @SchemaFieldDescription(
+        "The schema in which the data is encoded in the Kafka topic. "
```
Comment (andreigurau) [inline syndeo-template/src/main/java/com/google/cloud/syndeo/transforms/pubsub/SyndeoPubsubReadSchemaTransformProvider.java:204]: should be "Pubsub"

**Review by andreigurau — COMMENTED** [review]
should be "Pubsub"

File: syndeo-template/src/main/java/com/google/cloud/syndeo/transforms/pubsub/SyndeoPubsubReadSchemaTransformProvider.java [hunk 2]
```diff
@@ -178,12 +184,27 @@ public PCollectionRowTuple expand(PCollectionRowTuple input) {
...
   @AutoValue
   public abstract static class SyndeoPubsubReadSchemaTransformConfiguration {
+    @SchemaFieldDescription(
+        "The name of the topic to consume data from. If a topic is specified, Syndeo"
+            + " will create a new subscription for that topic and start consuming from that point. "
+            + "Format: projects/${PROJECT}/topics/${TOPIC}")
     public abstract @Nullable String getTopic();
 
+    @SchemaFieldDescription(
+        "The name of the subscription to consume data. "
```
Comment (mahagen) [inline syndeo-template/src/main/java/com/google/cloud/syndeo/transforms/pubsub/SyndeoPubsubReadSchemaTransformProvider.java:194]: Do we also want to mention that either a topic or a subscription should be provided? 

"The name of the subscription to consume data from. Either a topic or subscription must be provided. "

File: syndeo-template/src/main/java/com/google/cloud/syndeo/transforms/pubsub/SyndeoPubsubReadSchemaTransformProvider.java [hunk 3]
```diff
@@ -178,12 +184,27 @@ public PCollectionRowTuple expand(PCollectionRowTuple input) {
   @DefaultSchema(AutoValueSchema.class)
   @AutoValue
   public abstract static class SyndeoPubsubReadSchemaTransformConfiguration {
+    @SchemaFieldDescription(
+        "The name of the topic to consume data from. If a topic is specified, Syndeo"
+            + " will create a new subscription for that topic and start consuming from that point. "
```
Comment (mahagen) [inline syndeo-template/src/main/java/com/google/cloud/syndeo/transforms/pubsub/SyndeoPubsubReadSchemaTransformProvider.java:189]: Similar idea here too, add: 

"Either a topic or a subscription must be provided. "

File: syndeo-template/src/main/java/com/google/cloud/syndeo/transforms/pubsub/SyndeoPubsubWriteSchemaTransformProvider.java [hunk 4]
```diff
@@ -92,10 +110,17 @@ public PCollectionRowTuple expand(PCollectionRowTuple input) {
...
+  @DefaultSchema(AutoValueSchema.class)
   @AutoValue
   public abstract static class SyndeoPubsubWriteConfiguration {
+    @SchemaFieldDescription(
+        "The encoding format for the data stored in Pubsub. Valid options are: "
+            + VALID_FORMATS_STR)
     public abstract String getFormat();
 
+    @SchemaFieldDescription(
+        "The name of the topic to write data into. "
```
Comment (mahagen) [inline syndeo-template/src/main/java/com/google/cloud/syndeo/transforms/pubsub/SyndeoPubsubWriteSchemaTransformProvider.java:122]: The name of the topic to write data to.

**Review by mahagen — DISMISSED** [review]
Looks good, just some nits.

File: syndeo-template/src/main/java/com/google/cloud/syndeo/transforms/pubsub/SyndeoPubsubWriteSchemaTransformProvider.java (hunk 4 above)
Comment (pabloem) [reply syndeo-template/src/main/java/com/google/cloud/syndeo/transforms/pubsub/SyndeoPubsubWriteSchemaTransformProvider.java]: done!

File: syndeo-template/src/main/java/com/google/cloud/syndeo/transforms/pubsub/SyndeoPubsubReadSchemaTransformProvider.java (hunk 3 above)
Comment (pabloem) [reply syndeo-template/src/main/java/com/google/cloud/syndeo/transforms/pubsub/SyndeoPubsubReadSchemaTransformProvider.java:189]: done!

File: syndeo-template/src/main/java/com/google/cloud/syndeo/transforms/pubsub/SyndeoPubsubReadSchemaTransformProvider.java (hunk 2 above)
Comment (pabloem) [reply syndeo-template/src/main/java/com/google/cloud/syndeo/transforms/pubsub/SyndeoPubsubReadSchemaTransformProvider.java:195]: done!

File: syndeo-template/src/main/java/com/google/cloud/syndeo/transforms/pubsub/SyndeoPubsubReadSchemaTransformProvider.java (hunk 1 above)
Comment (pabloem) [reply syndeo-template/src/main/java/com/google/cloud/syndeo/transforms/pubsub/SyndeoPubsubReadSchemaTransformProvider.java]: done!

**Review by andreigurau — APPROVED** [review]

**Pull request merged**